name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/MyDividends24Uploader.java

```diff
@@ -50,21 +53,37 @@ public List<Pair<Integer, String>> getPortfolios() throws IOException
     }
 
     @SuppressWarnings("unchecked")
-    public void upload(Client client, CurrencyConverter converter, String portfolioId) throws IOException
+    public void upload(Client client, CurrencyConverter converter, String portfolioId, Portfolio ppPortfolio)
+                    throws IOException
     {
-        ClientSnapshot snapshot = ClientSnapshot.create(client, converter, LocalDate.now());
-        PortfolioSnapshot portfolio = snapshot.getJointPortfolio();
+
+        Stream<PortfolioTransaction> stream;
+
+        if (ppPortfolio != null)
+        {
+            // Case: one portfolio of PP is selected
+            stream = ppPortfolio.getTransactions().stream();
+        }
+        else
+        {
+            // Case: all portfolios of PP is selected
+            ClientSnapshot snapshot = ClientSnapshot.create(client, converter, LocalDate.now());
+            PortfolioSnapshot portfolio = snapshot.getJointPortfolio();
+            stream = portfolio.getPortfolio().getTransactions().stream();
```

Comment:
I believe this can be simplified to

```java
stream = client.getPortfolios().stream().flatMap(p -> p.getTransactions().stream());
```

Because you are interested only in the transactions (see the code further below).
There is no need to calculate a snapshot then (which also means you do not need the CurrencyConverter here).

And: wording of the comment: "case: all portfolios **are** selected"